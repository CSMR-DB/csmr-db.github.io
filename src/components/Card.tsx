import styled, { StyledComponent } from 'styled-components'
// tslint:disable-next-line: no-implicit-dependencies
import { BackgroundColorProperty } from 'csstype'

export const Card: StyledComponent<'article', any, {}, never> = styled.article`
  position: relative;
  border-radius: 0.25rem;
  display: grid;
  grid-template-areas: 'header' 'body' 'footer';
  grid-template-rows: auto 1fr auto;
  overflow: hidden;
  background: white;
  box-shadow: 0 0.075rem 0.075rem rgba(0, 0, 0, 0.1),
    0 0.125rem 0.125rem rgba(0, 0, 0, 0.25);
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);

  &:hover {
    box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.25),
      0 0.25rem 0.25rem rgba(0, 0, 0, 0.2);
    transform: translateY(-0.25rem);
  }

  & h1 {
    font-size: 1.5em;
    margin: 0;
  }

  & h2 {
    font-size: 1.1em;
    font-style: italic;
    margin: 0;
    color: #aaa;
    margin-bottom: 1rem;
  }

  & p {
    margin: 0;
    text-align: justify;
  }

  & header {
    margin-bottom: 2em;
  }
`

export const CardHeader: StyledComponent<
  'header',
  any,
  {},
  never
> = styled.header`
  margin-bottom: 2em;
  max-height: 18rem;
  overflow: hidden;
  grid-area: 'header';
`

export const CardHeaderText: StyledComponent<
  'div',
  any,
  {},
  never
> = styled.div`
  padding-left: 2em;
`

interface ICardHeaderIconWrapperProps {
  backgroundColor: BackgroundColorProperty
}

export const CardHeaderIconWrapper: StyledComponent<
  'div',
  any,
  ICardHeaderIconWrapperProps,
  never
> = styled.div`
  border-radius: 999px;
  overflow: hidden;
  max-width: 4em;
  width: 100%;
  max-height: 4em;
  height: 100%;
  background: ${(props: ICardHeaderIconWrapperProps): BackgroundColorProperty =>
    props.backgroundColor || '#DDD'};
  padding: 0.5em;
`

export const CardBody: StyledComponent<'main', any, {}, never> = styled.main`
  position: relative;
  padding: 2em;
  grid-area: 'body';
`

export const CardFooter: StyledComponent<
  'footer',
  any,
  {},
  never
> = styled.footer`
  background: #eee;
  padding: 2em;
  border-top: 1px solid grey;
  grid-area: 'footer';

  & span {
    padding: 0 0.25em;
  }
`
