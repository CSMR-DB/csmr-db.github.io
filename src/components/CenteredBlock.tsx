import styled, { StyledComponent } from 'styled-components'
import { WidthProperty } from 'csstype'

interface ICenteredBlockProps {
  maxWidth?: WidthProperty<1>
}

export const CenteredBlock: StyledComponent<
  'div',
  any,
  ICenteredBlockProps,
  never
> = styled.div`
  max-width: ${({
    maxWidth = '48rem',
  }: ICenteredBlockProps): WidthProperty<1> => maxWidth};
  margin: 0 auto;
`
