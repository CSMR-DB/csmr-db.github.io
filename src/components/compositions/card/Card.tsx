import React, { ReactNode } from 'react'
import styled, {
  StyledComponent,
  DefaultTheme,
  css,
  FlattenSimpleInterpolation,
} from 'styled-components'
import { Tween } from 'react-gsap'

export interface ICardProps {
  theme?: DefaultTheme
  children: ReactNode
  index?: number
}

export const StyledCard: StyledComponent<
  'article',
  any,
  ICardProps,
  never
> = styled.article`
  position: relative;
  height: 100%;
  border-radius: 0.25rem;
  display: grid;
  grid-template-areas: 'header' 'body' 'footer';
  grid-template-rows: auto 1fr auto;
  overflow: hidden;
  box-shadow: 0 0.075rem 0.075rem rgba(0, 0, 0, 0.1),
    0 0.125rem 0.125rem rgba(0, 0, 0, 0.25);
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);

  & header img {
    transition: all 0.2s ease-in-out !important;
  }

  & h1 {
    font-size: 1.1rem;
    margin: 0;
  }

  & h2 {
    font-size: 1rem;
    font-style: italic;
    margin: 0;
    color: #aaa;
    margin-bottom: 1rem;
  }

  & p {
    text-align: left;
  }

  ${({ theme }: ICardProps): FlattenSimpleInterpolation => css`
    background: ${theme!.palette.light.normal};

    @media ${theme!.breakpoints.min.smartphone} {
      text-align: justify;
    }

    @media ${theme!.breakpoints.min.desktop} {
      &:hover {
        box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.25),
          0 0.25rem 0.25rem rgba(0, 0, 0, 0.2);

        & header img {
          transform: scale(1.1);
        }
      }
    }
  `}
`

export function Card({ children, index = 0 }: ICardProps): JSX.Element {
  return (
    <Tween
      from={{ scale: 0.95, opacity: 0 }}
      duration={0.5}
      delay={0.25 + index * 0.125}
      ease="back"
    >
      <StyledCard>{children}</StyledCard>
    </Tween>
  )
}
