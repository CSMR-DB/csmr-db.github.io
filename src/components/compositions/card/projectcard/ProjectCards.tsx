import React from 'react'

import { IProjectsAllMarkdownRemark, IProjectsEdge } from '../../../../types/interfaces'

import { ProjectCard } from './ProjectCard'

export function ProjectCards({
  edges,
}: IProjectsAllMarkdownRemark): JSX.Element {
  return (
    <>
      {edges.map(
        (
          { node: { frontmatter, timeToRead } }: IProjectsEdge,
          i: number
        ): JSX.Element => (
          <ProjectCard
            title={frontmatter.title}
            tags={frontmatter.tags}
            video={frontmatter.featuredVideo}
            image={frontmatter.featuredImage}
            body={frontmatter.excerpt}
            timeToRead={timeToRead}
            path={frontmatter.path}
            date={frontmatter.date}
            key={i}
            index={i}
          />
        )
      )}
    </>
  )
}
