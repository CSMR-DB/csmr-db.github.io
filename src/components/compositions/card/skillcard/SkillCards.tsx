import React from 'react'

import {
  ISkillsetAllMarkdownRemark,
  ISkillsetEdge,
} from '../../../../types/interfaces'
import { routeGenerator } from '../../../../utils/routeGenerator'

import { SkillCard } from './SkillCard'

export function SkillCards({ edges }: ISkillsetAllMarkdownRemark): JSX.Element {
  return (
    <>
      {edges.map(
        (
          { node: { frontmatter, fileAbsolutePath } }: ISkillsetEdge,
          i: number
        ): JSX.Element => (
          <SkillCard
            level={frontmatter.level || 0}
            title={frontmatter.title}
            description={frontmatter.excerpt}
            time={frontmatter.time}
            key={i}
            index={i}
            skillColor={frontmatter.skillColor || 'black'}
            fileAbsolutePath={routeGenerator(fileAbsolutePath)}
          />
        )
      )}
    </>
  )
}
