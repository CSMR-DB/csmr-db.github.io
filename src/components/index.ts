export { Absolute } from './Absolute'
export { ArticleBody } from './ArticleBody'
export { ArticleTags } from './ArticleTags'
export { Blend } from './Blend'
export {
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardHeaderFeaturedImage,
  CardHeaderIconWrapper,
  CardHeaderText,
} from './Card'
export { CardHeaderImageOrVideo } from './CardHeaderImageOrVideo'
export { CenteredBlock } from './CenteredBlock'
export { ClippedBtn } from './ClippedBtn'
export { ContentSeparator } from './ContentSeparator'
export { CTAButton } from './CTAButton'
export { DateFormatted } from './DateFormatted'
export { DesktopNav } from './DesktopNav'
export { DynamicIcon } from './DynamicIcon'
export { DynamicImage } from './DynamicImage'
export { ExperienceCard } from './ExperienceCard'
export { Excerpt } from './Excerpt'
export { Filter } from './Filter'
export { Fixed } from './Fixed'
export { Flex } from './Flex'
export { Footer } from './Footer'
export { Grid } from './Grid'
export { Header } from './Header'
export { Heart } from './Heart'
export { Image } from './Image'
export { ImageContainerCircle } from './ImageContainerCircle'
export { ImageDot } from './ImageDot'
export { Layout } from './Layout'
export { Logo } from './Logo'
export { MenuTile } from './MenuTile'
export { ProjectCard } from './ProjectCard'
export { ReadingTime } from './ReadingTime'
export { Relative } from './Relative'
export {
  ScrollIconWrapper,
  AnimatedScroller,
  ScrollText,
} from './ScrollIndicator'
export { SEO } from './SEO'
export { SkillCard } from './SkillCard'
export { SocialButton } from './SocialButton'
export { StyledIFrame } from './StyledIFrame'
export { StyledLink, StyledA } from './StyledLink'
export { WASDNav } from './WASDNav'
