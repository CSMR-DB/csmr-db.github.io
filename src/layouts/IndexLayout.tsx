import React from 'react';
import { graphql, useStaticQuery } from 'gatsby';
import { Layout, SEO } from '../components';

export interface ISiteMetadata {
  site: {
    siteMetadata: {
      title: string
      description: string
      author: string
    }
  }
}

export function IndexLayout(): JSX.Element {
  // tslint:disable-next-line: no-void-expression
  const data: ISiteMetadata = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
          description
          author
        }
      }
    }
  `)

  return (
    <>
      <SEO title={`Homepage â€¢ ${data.site.siteMetadata.description}`} description={data.site.siteMetadata.description} />
      <Layout isLandingPage />
    </>
  )
}