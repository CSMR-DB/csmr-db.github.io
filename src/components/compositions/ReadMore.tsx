import React from 'react'
import styled, {
  StyledComponent,
  DefaultTheme,
  css,
  FlattenSimpleInterpolation,
} from 'styled-components'

import { Card } from './card/Card'
import { CardBody } from './card/CardBody'
import { StyledLink } from '../StyledLink'

export interface IReadMoreProps {
  theme?: DefaultTheme
  title: string
  link: {
    path: string
    text: string
  }
  $_testid?: string
}

export const StyledReadMore: StyledComponent<
  'div',
  any,
  Pick<IReadMoreProps, 'theme'>,
  never
> = styled.div`
  display: flex;

  ${({ theme }: IReadMoreProps): FlattenSimpleInterpolation => css`
    flex-direction: column;
    gap: 1rem;

    @media ${theme!.breakpoints.min.smartphone} {
      flex-direction: row;
      justify-content: space-between;
      align-items: baseline;
    }
  `}
`

export function ReadMore({
  title,
  link,
  $_testid,
}: IReadMoreProps): JSX.Element {
  return (
    <Card $_testid={$_testid}>
      <CardBody>
        <StyledReadMore>
          <h1>{title}</h1>
          <StyledLink to={link.path} $_testid={'ReadMoreLink'}>
            {link.text}
          </StyledLink>
        </StyledReadMore>
      </CardBody>
    </Card>
  )
}
