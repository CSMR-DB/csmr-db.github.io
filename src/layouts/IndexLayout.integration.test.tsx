import React, { Component } from 'react'
import { ThemeProvider } from 'styled-components'
import { create, ReactTestRenderer } from 'react-test-renderer'
import { mount, ReactWrapper } from 'enzyme'

import { StaticDataManager } from '../data/StaticDataManager'

import { IndexLayout } from './IndexLayout'
import { Layout } from '../components/Layout'
import { LiftedUpWrapper } from '../components/LiftedUpWrapper'
import { CenteredBlock } from '../components/CenteredBlock'
import { Grid } from '../components/Grid'
import { ReadMore } from '../components/compositions/ReadMore'
import { Card } from '../components/compositions/card/Card'
import { SkillCards } from '../components/compositions/card/skillcard/SkillCards'
import { ProjectCards } from '../components/compositions/card/projectcard/ProjectCards'

describe('IndexLayout', (): void => {
  it('should match snapshot', (): void => {
    const comp: ReactTestRenderer = create(
      <ThemeProvider theme={StaticDataManager.theme}>
        <IndexLayout
          skillsetData={StaticDataManager.skillsetData.top}
          projectsData={StaticDataManager.projectsData.top}
        />
      </ThemeProvider>
    )

    expect(comp).toMatchSnapshot()
  })

  it('should render internally', (): void => {
    const comp: ReactWrapper<any, Readonly<{}>, Component<{}, {}, any>> = mount(
      <ThemeProvider theme={StaticDataManager.theme}>
        <IndexLayout
          skillsetData={StaticDataManager.skillsetData.top}
          projectsData={StaticDataManager.projectsData.top}
        />
      </ThemeProvider>
    )

    expect(comp.find(Layout).props().isLandingPage).toEqual(true)
    expect(comp.find(Layout).find(LiftedUpWrapper).length).toEqual(1)
    expect(
      comp.find(Layout).find(LiftedUpWrapper).find(CenteredBlock).length
    ).toEqual(1)
    expect(
      comp.find(Layout).find(LiftedUpWrapper).find(CenteredBlock).props()
        .$maxWidth
    ).toEqual('110rem')

    expect(
      comp.find(Layout).find(LiftedUpWrapper).find(CenteredBlock).find(Grid)
        .length
    ).toEqual(4)
    expect(
      comp
        .find(Layout)
        .find(LiftedUpWrapper)
        .find(CenteredBlock)
        .find(Grid)
        .at(1)
        .find(ReadMore).length
    ).toEqual(1)
    expect(
      comp
        .find(Layout)
        .find(LiftedUpWrapper)
        .find(CenteredBlock)
        .find(Grid)
        .at(1)
        .find(ReadMore)
        .props().link
    ).toEqual({
      path: '/about',
      text: 'my story',
    })
    expect(
      comp
        .find(Layout)
        .find(LiftedUpWrapper)
        .find(CenteredBlock)
        .find(Grid)
        .at(1)
        .find(Card).length
    ).toEqual(2)
    expect(
      comp
        .find(Layout)
        .find(LiftedUpWrapper)
        .find(CenteredBlock)
        .find(Grid)
        .at(1)
        .find(Card)
        .at(1)
        .find('h1')
        .text()
    ).toEqual('TL;WR')

    expect(
      comp
        .find(Layout)
        .find(LiftedUpWrapper)
        .find(CenteredBlock)
        .find(Grid)
        .at(2)
        .find(ReadMore).length
    ).toEqual(1)
    expect(
      comp
        .find(Layout)
        .find(LiftedUpWrapper)
        .find(CenteredBlock)
        .find(Grid)
        .at(2)
        .find(ReadMore)
        .props().link
    ).toEqual({
      path: '/skillset',
      text: 'my toolbelt',
    })
    expect(
      comp
        .find(Layout)
        .find(LiftedUpWrapper)
        .find(CenteredBlock)
        .find(Grid)
        .at(2)
        .find(SkillCards).length
    ).toEqual(1)

    expect(
      comp
        .find(Layout)
        .find(LiftedUpWrapper)
        .find(CenteredBlock)
        .find(Grid)
        .at(3)
        .find(ReadMore).length
    ).toEqual(1)
    expect(
      comp
        .find(Layout)
        .find(LiftedUpWrapper)
        .find(CenteredBlock)
        .find(Grid)
        .at(3)
        .find(ReadMore)
        .props().link
    ).toEqual({
      path: '/projects',
      text: 'my work / experiments',
    })
    expect(
      comp
        .find(Layout)
        .find(LiftedUpWrapper)
        .find(CenteredBlock)
        .find(Grid)
        .at(3)
        .find(ProjectCards).length
    ).toEqual(1)
  })
})
