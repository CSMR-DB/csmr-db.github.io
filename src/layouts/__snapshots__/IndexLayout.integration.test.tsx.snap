// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexLayout should match snapshot 1`] = `
.c1 {
  margin: 0 auto;
  padding: 1rem;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #f00;
}

.c1 a:hover {
  color: #d00;
}

.c0 {
  min-height: 100vh;
  background: transparent;
}

.c11 {
  grid-area: 'header';
}

.c8 {
  grid-area: 'body';
  position: relative;
  padding: 2rem;
}

.c12 {
  padding: 2rem 2rem 0 2rem;
}

.c12 > h6 {
  margin-bottom: 0;
}

.c16 {
  fill: red;
  width: 1rem;
  height: 1rem;
}

.c19 {
  fill: blue;
  width: 1rem;
  height: 1rem;
}

.c13 {
  -webkit-transition: all ease 0.5s;
  transition: all ease 0.5s;
  position: absolute;
  bottom: 1rem;
  right: 1rem;
  width: 100%;
  max-width: 8rem;
}

.c14 {
  padding: 2rem 2rem 0 2rem;
}

.c14 h1 {
  font-size: 1rem;
  font-weight: 100;
  text-align: right;
  -webkit-align-self: normal;
  -ms-flex-item-align: normal;
  align-self: normal;
}

.c14 h2 {
  font-size: 0.8rem;
  font-weight: 100;
}

.c18 {
  grid-area: 'footer';
  padding: 0 2rem 2rem;
}

.c15 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  height: 100%;
  width: 100%;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: initial;
  -ms-flex-wrap: initial;
  flex-wrap: initial;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c10 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  color: #f00 !important;
}

.c10:hover {
  color: #d00;
}

.c10.active {
  color: #b00;
}

.c17 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  color: red !important;
}

.c17:hover {
  color: #d00;
}

.c17.active {
  color: #b00;
}

.c20 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  color: blue !important;
}

.c20:hover {
  color: #d00;
}

.c20.active {
  color: #b00;
}

.c7 {
  position: relative;
  height: 100%;
  border-radius: 0.25rem;
  display: grid;
  grid-template-areas: 'header' 'body' 'footer';
  grid-template-rows: auto 1fr auto;
  overflow: hidden;
  box-shadow: 0 0.075rem 0.075rem rgba(0,0,0,0.1), 0 0.125rem 0.125rem rgba(0,0,0,0.25);
  -webkit-transition: all 0.3s cubic-bezier(0.25,0.8,0.25,1);
  transition: all 0.3s cubic-bezier(0.25,0.8,0.25,1);
  background: #fff;
}

.c7 header img {
  -webkit-transition: all 0.2s ease-in-out !important;
  transition: all 0.2s ease-in-out !important;
}

.c7 h1 {
  font-size: 1.1rem;
  margin: 0;
}

.c7 h2 {
  font-size: 1rem;
  font-style: italic;
  margin: 0;
  color: #aaa;
  margin-bottom: 1rem;
}

.c7 p {
  text-align: left;
}

.c21 {
  text-align: left !important;
}

.c2 {
  margin: 0 auto;
  padding: 0 1rem;
}

.c3 {
  padding-bottom: 4rem;
}

.c4 {
  margin: 0 auto;
  max-width: 110rem;
}

.c5 {
  display: grid;
  margin: 0 auto;
  height: auto;
  max-width: auto;
  gap: calc(1rem / 2);
  grid-template-columns: repeat(auto-fill,minmax(16rem,1fr));
  grid-template-rows: auto;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
}

.c6 {
  display: grid;
  margin: 0 auto;
  height: auto;
  max-width: auto;
  gap: calc(1rem / 2);
  grid-template-columns: repeat(auto-fill,minmax(16rem,1fr));
  grid-template-rows: auto;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

@media only screen and (min-width:600px) {
  .c1 {
    padding: 4rem 1rem;
  }
}

@media only screen and (min-width:600px) {
  .c7 {
    text-align: justify;
  }
}

@media only screen and (min-width:1200px) {
  .c7:hover {
    box-shadow: 0 0.125rem 0.25rem rgba(0,0,0,0.25), 0 0.25rem 0.25rem rgba(0,0,0,0.2);
  }

  .c7:hover header img {
    -webkit-transform: scale(1.1);
    -ms-transform: scale(1.1);
    transform: scale(1.1);
  }
}

@media only screen and (min-width:600px) {
  .c2 {
    margin: -8rem auto 0;
  }
}

@media only screen and (min-width:600px) {
  .c5 {
    grid-template-columns: repeat(3,1fr);
    gap: 1rem;
  }
}

@media only screen and (min-width:600px) {
  .c6 {
    grid-template-columns: auto;
    gap: 1rem;
  }
}

@media only screen and (min-width:600px) {
  .c9 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: baseline;
    -webkit-box-align: baseline;
    -ms-flex-align: baseline;
    align-items: baseline;
  }
}

<div
  className="c0"
>
  <main
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          <div
            className="c5"
          >
            <div>
              <div
                className="c6"
              >
                <article
                  className="c7"
                >
                  <main
                    className="c8"
                  >
                    <div
                      className="c9"
                    >
                      <h1>
                        Why I make things
                      </h1>
                      <a
                        className="c10"
                        href="/about"
                      >
                        my story
                      </a>
                    </div>
                  </main>
                </article>
                <article
                  className="c7"
                >
                  <header
                    className="c11"
                  >
                    <div
                      className="c12"
                    >
                      <h1>
                        TL;WR
                      </h1>
                    </div>
                  </header>
                  <main
                    className="c8"
                  >
                    <p>
                      I find the ability to create something out of nothing but some text and symbols just absolutely fantastic.
                    </p>
                    <p>
                      I learned about programming at college, where it was just part of the entire curriculum. During my time there, I taught myself mostly CSS & HTML, but also a decent amount of JavaScript. I played with React, Vue, MeteorJS, CSS preprocessors like Stylus and SASS.
                    </p>
                    <p>
                      Post-college I started experimenting with Back-end NodeJS, GraphQL, MongoDB and, most importantly, TypeScript. I also dipped my toes in the Angular framework and messed with some Svelte. But whatever framework / library you want to work with honestly doesn't matter. You can learn to use & master a framework, yet understanding the underlying language is what matters most.
                    </p>
                  </main>
                </article>
              </div>
            </div>
            <div>
              <div
                className="c6"
              >
                <article
                  className="c7"
                >
                  <main
                    className="c8"
                  >
                    <div
                      className="c9"
                    >
                      <h1>
                        How I make things
                      </h1>
                      <a
                        className="c10"
                        href="/skillset"
                      >
                        my toolbelt
                      </a>
                    </div>
                  </main>
                </article>
                <article
                  className="c7"
                >
                  <div
                    className="c13 backdrop"
                  >
                    
                  </div>
                  <header
                    className="c11"
                  >
                    <div
                      className="c14"
                    >
                      <div
                        className="c15"
                      >
                        <svg
                          className="c16"
                          viewBox="0 0 315 342"
                        >
                          <g
                            id="heart"
                          >
                            <path
                              d="M0 200 v-200 h200 a100,100 90 0,1 0,200 a100,100 90 0,1 -200,0 z"
                            />
                          </g>
                        </svg>
                        <h1>
                          <a
                            className="c17"
                            href="/skillset/react"
                          >
                            React
                          </a>
                        </h1>
                      </div>
                    </div>
                  </header>
                  <main
                    className="c8"
                  >
                    <p>
                      I love JSX
                    </p>
                  </main>
                  <footer
                    className="c18"
                  >
                    <p>
                      (time spent:
                       
                      <span>
                        0
                      </span>
                       
                      hours)
                    </p>
                  </footer>
                </article>
                <article
                  className="c7"
                >
                  <div
                    className="c13 backdrop"
                  >
                    
                  </div>
                  <header
                    className="c11"
                  >
                    <div
                      className="c14"
                    >
                      <div
                        className="c15"
                      >
                        <svg
                          className="c19"
                          viewBox="0 0 315 342"
                        >
                          <g
                            id="heart"
                          >
                            <path
                              d="M0 200 v-200 h200 a100,100 90 0,1 0,200 a100,100 90 0,1 -200,0 z"
                            />
                          </g>
                        </svg>
                        <h1>
                          <a
                            className="c20"
                            href="/skillset/jest"
                          >
                            Jest
                          </a>
                        </h1>
                      </div>
                    </div>
                  </header>
                  <main
                    className="c8"
                  >
                    <p>
                      It Jest needs to be done
                    </p>
                  </main>
                  <footer
                    className="c18"
                  >
                    <p>
                      (time spent:
                       
                      <span>
                        0
                      </span>
                       
                      hours)
                    </p>
                  </footer>
                </article>
              </div>
            </div>
            <div>
              <div
                className="c6"
              >
                <article
                  className="c7"
                >
                  <main
                    className="c8"
                  >
                    <div
                      className="c9"
                    >
                      <h1>
                        What I have made
                      </h1>
                      <a
                        className="c10"
                        href="/projects"
                      >
                        my work / experiments
                      </a>
                    </div>
                  </main>
                </article>
                <article
                  className="c7"
                >
                  <header
                    className="c11"
                  >
                    <div
                      className="c12"
                    >
                      <h1>
                        <a
                          className="c10"
                          href="/projects/circle"
                        >
                          Circle
                        </a>
                      </h1>
                      <h6>
                        <span>
                          08 Jan 1994
                        </span>
                      </h6>
                    </div>
                  </header>
                  <main
                    className="c8"
                  >
                    <p>
                      I drew a fantastic circle once
                    </p>
                  </main>
                  <footer
                    className="c18"
                  >
                    <p
                      className="c21"
                    >
                      Tags:
                       
                      <span>
                        <a
                          className="c10"
                          href="/skillset/pencil"
                        >
                          Pencil
                        </a>
                        , 
                      </span>
                      <span>
                        <a
                          className="c10"
                          href="/skillset/shapes"
                        >
                          Shapes
                        </a>
                        , 
                      </span>
                      <span>
                        <a
                          className="c10"
                          href="/skillset/drawing"
                        >
                          Drawing
                        </a>
                        
                      </span>
                    </p>
                  </footer>
                </article>
                <article
                  className="c7"
                >
                  <header
                    className="c11"
                  >
                    <div
                      className="c12"
                    >
                      <h1>
                        <a
                          className="c10"
                          href="/projects/portfolio"
                        >
                          Portfolio
                        </a>
                      </h1>
                      <h6>
                        <span>
                          08 Jan 1994
                        </span>
                      </h6>
                    </div>
                  </header>
                  <main
                    className="c8"
                  >
                    <p>
                      Gatsby, TypeScript, React, GraphQL, GSAP, Styled-Components, Jest, Enzyme
                    </p>
                  </main>
                  <footer
                    className="c18"
                  >
                    <p
                      className="c21"
                    >
                      Tags:
                       
                      <span>
                        <a
                          className="c10"
                          href="/skillset/gatsby"
                        >
                          Gatsby
                        </a>
                        , 
                      </span>
                      <span>
                        <a
                          className="c10"
                          href="/skillset/typescript"
                        >
                          TypeScript
                        </a>
                        , 
                      </span>
                      <span>
                        <a
                          className="c10"
                          href="/skillset/react"
                        >
                          React
                        </a>
                        , 
                      </span>
                      <span>
                        <a
                          className="c10"
                          href="/skillset/graphql"
                        >
                          GraphQL
                        </a>
                        , 
                      </span>
                      <span>
                        <a
                          className="c10"
                          href="/skillset/gsap"
                        >
                          GSAP
                        </a>
                        , 
                      </span>
                      <span>
                        <a
                          className="c10"
                          href="/skillset/styled-components"
                        >
                          Styled-Components
                        </a>
                        , 
                      </span>
                      <span>
                        <a
                          className="c10"
                          href="/skillset/jest"
                        >
                          Jest
                        </a>
                        , 
                      </span>
                      <span>
                        <a
                          className="c10"
                          href="/skillset/enzyme"
                        >
                          Enzyme
                        </a>
                        
                      </span>
                    </p>
                  </footer>
                </article>
                <article
                  className="c7"
                >
                  <header
                    className="c11"
                  >
                    <div
                      className="c12"
                    >
                      <h1>
                        <a
                          className="c10"
                          href="/projects/portfolio"
                        >
                          Loopy
                        </a>
                      </h1>
                      <h6>
                        <span>
                          08 Jan 1994
                        </span>
                      </h6>
                    </div>
                  </header>
                  <main
                    className="c8"
                  >
                    <p>
                      I made a fantastic hula-loop once
                    </p>
                  </main>
                  <footer
                    className="c18"
                  >
                    <p
                      className="c21"
                    >
                      Tags:
                       
                      <span>
                        <a
                          className="c10"
                          href="/skillset/hula"
                        >
                          Hula
                        </a>
                        , 
                      </span>
                      <span>
                        <a
                          className="c10"
                          href="/skillset/loop"
                        >
                          Loop
                        </a>
                        , 
                      </span>
                      <span>
                        <a
                          className="c10"
                          href="/skillset/perfection"
                        >
                          Perfection
                        </a>
                        
                      </span>
                    </p>
                  </footer>
                </article>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
